### PR TITLE
Revised AI engineering docs (2)

### PR DESCRIPTION
**Pull request by dominicchapman**
New "evaluation" content:
- Updated workflow language from "Measure" to "Evaluate" to better reflect our approach
- Reorganized evaluation content into a dedicated section with six focused pages (overview, setup, write evaluations, flags & experiments, run evaluations, analyze results)

Other changes:
- Concepts: Added definitions for flags and experiments; integrated AI capability architecture spectrum (single-turn → workflows → single-agent → multi-agent)
- Create: De-emphasized experimental prompt management features while clarifying Axiom's current focus on evaluation and observability; added references to Vercel AI SDK examples and Mastra as framework alternatives
- Iterate: Complete rewrite introducing the systematic improvement loop; added sections on user feedback capture and domain expert annotation workflows (marked as coming soon); reorganized failure categorization by severity for better prioritization
- Quickstart: Updated to reference evaluation framework and CLI authentication; improved "What's next" guidance

### CODE REVIEW

File: ai-engineering/concepts.mdx [hunk 1]
```diff
@@ -33,7 +33,12 @@ The concepts in AI engineering are best understood within the context of the dev
...
 
-Capabilities exist on a spectrum of complexity. They can be a simple, single-step function (for example, classifying a support ticket’s intent) or evolve into a sophisticated, multi-step agent that uses reasoning and tools to achieve a goal (for example, orchestrating a complete customer support resolution).
+Capabilities exist on a spectrum of complexity, ranging from simple to sophisticated architectures:
+
+- **Single-turn model interactions**: A single prompt and response, such as classifying a support ticket’s intent or summarizing a document.
+- **Workflows**: Multi-step processes where each step’s output feeds into the next, such as research → analysis → report generation.
+- **Single-agent**: An agent that can reasons and make decisions to accomplish a goal, such as a customer support agent that can search documentation, check order status, and draft responses.
+- **Multi-agent**: Multiple specialized agents collaborating to solve complex problems, such as software engineering through architectural planning, coding, testing, and review.
```
Comment (c-ehrlich) [inline ai-engineering/concepts.mdx:41]: I still don't love talking about multi-agent in this way, because every real-world implementation of multi-agents I've seen was just workflows and/or subagents

- Workflows have steps
- Agents have tools
- A workflow can be a step
- A workflow can be a tool
- An agent can be a step
- An agent can be a tool 

But also definitely don't want to block on this kind of discussion :)

File: ai-engineering/concepts.mdx [hunk 2]
```diff
@@ -55,17 +60,25 @@ Ground truth is the validated, expert-approved correct output for a given input.
 
 Annotations are expert-provided labels, corrections, or outputs added to records to establish or refine ground truth.
 
-### Grader
+### Scorer
+
+A scorer is a function that evaluates a capability’s output. It programmatically assesses quality by comparing the generated output against ground truth or other criteria, returning a score or judgment. Scorers are the reusable, atomic scoring logic used in all forms of evaluation.
```
Comment (c-ehrlich) [inline ai-engineering/concepts.mdx]:
```suggestion
A scorer is a function that evaluates a capability’s output. It programmatically assesses quality by comparing the generated output against ground truth or other criteria, returning a score.
```
Not sure what the other words add here

File: ai-engineering/concepts.mdx [hunk 3]
```diff
@@ -55,17 +60,25 @@ Ground truth is the validated, expert-approved correct output for a given input.
...
+
+### Flag
+
+A flag is a configuration parameter that controls how your AI capability behaves. Flags let you parameterize aspects like model choice, temperature, prompting strategies, or retrieval approaches. By defining flags, you can run experiments to compare different configurations and systematically determine which approach performs best.
```
Comment (c-ehrlich) [inline ai-engineering/concepts.mdx]:
```suggestion
A flag is a configuration parameter that controls how your AI capability behaves. Flags let you parameterize aspects like model choice, prompting strategies, or retrieval approaches. By defining flags, you can run experiments to compare different configurations and systematically determine which approach performs best.
```
not a fan of how much we talk about temperature, and it's implied by "prompting strategies" anyway

File: ai-engineering/concepts.mdx [hunk 4]
```diff
@@ -55,17 +60,25 @@ Ground truth is the validated, expert-approved correct output for a given input.
...
+
+### Experiment
 
-A grader is a function that scores a capability’s output. It programmatically assesses quality by comparing the generated output against ground truth or other criteria, returning a score or judgment. Graders are the reusable, atomic scoring logic used in all forms of evaluation.
+An experiment is an evaluation run with a specific set of flag values. By running multiple experiments with different flag configurations, you can compare performance across different models, prompts, or strategies to find the optimal setup for your capability.
 
-### Evaluator (eval)
+### Evaluator or "eval"
```
Comment (c-ehrlich) [inline ai-engineering/concepts.mdx]: when do we use Evaluator vs Evaluation? an Evaluation is when you run an Evaluator once? i hate that "eval" conflates the two quite often, but not much we can do about that probably.

File: ai-engineering/overview.mdx [hunk 5]
```diff
@@ -1,29 +1,27 @@
 ---
 title: "AI engineering overview"
-description: "Introduction to Axiom’s methodology for designing, evaluating, monitoring, and iterating generative-AI capabilities."
+description: "Build increasingly sophisticated AI capabilities with confidence using systematic evaluation and observability."
```
Comment (c-ehrlich) [inline ai-engineering/overview.mdx:3]:
```suggestion
description: "Build AI capabilities with confidence using systematic evaluation and observability."
```
is "increasingly sophisticated" necessary here?

File: ai-engineering/overview.mdx [hunk 6]
```diff
@@ -1,29 +1,27 @@
...
-* **Iterate**: Use insights from production to refine prompts, augment reference datasets, and improve the capability over time.
+* **Create**: Prototype your AI capability using any framework. TypeScript-based frameworks like Vercel AI SDK integrate most seamlessly with Axiom's tooling. As you build, gather reference examples to establish ground truth for evaluation.
+* **Evaluate**: Systematically test your capability’s performance against reference data using custom scorers to measure accuracy, quality, and cost. Use Axiom's evaluation framework to run experiments with different configurations and track improvements over time.
+* **Observe**: Deploy your capability and collect rich telemetry on every LLM call and tool execution. Use online evaluations to monitor for performance degradation and discover edge cases in production.
+* **Iterate**: Use insights from production monitoring and evaluation results to refine prompts, augment reference datasets, and improve the capability. Run new evaluations to verify improvements before deploying changes.
```
Comment (c-ehrlich) [inline ai-engineering/overview.mdx:24]: Love this

File: ai-engineering/quickstart.mdx [hunk 7]
```diff
@@ -214,7 +216,12 @@ ANTHROPIC_API_KEY=""
 Enter the API keys for the LLMs you want to work with.
 </Info>
 
+<Note>
+To run evaluations, you’ll need to authenticate the Axiom CLI. See [Setup and authentication](/ai-engineering/evaluate/setup) for details on using `axiom auth login`.
+</Note>
```
Comment (c-ehrlich) [inline ai-engineering/quickstart.mdx:221]: This is more for the dev team i guess, but we absolutely need to allow "true offline" evals that just report to console or a file, and don't require auth.

File: ai-engineering/evaluate/analyze-results.mdx [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+## Experiment with flags
+
+Flags let you test multiple configurations systematically. Run several experiments:
+
+```bash
+axiom eval --flag.model=gpt-4o-mini --flag.temperature=0.7
+axiom eval --flag.model=gpt-4o-mini --flag.temperature=0.2
+axiom eval --flag.model=gpt-4o --flag.temperature=0.7
+axiom eval --flag.model=gpt-4o --flag.temperature=0.2
```
Comment (c-ehrlich) [inline ai-engineering/evaluate/analyze-results.mdx]: already mentioned this somewhere else, but i think we should find a better example than temperature

File: ai-engineering/evaluate/analyze-results.mdx [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+```
+
+Compare all four runs in the Console to find the configuration that best balances quality, cost, and latency for your requirements.
```
Comment (c-ehrlich) [inline ai-engineering/evaluate/analyze-results.mdx:100]: Side note: we're going to need a good way of doing this 😅

File: ai-engineering/evaluate/flags-experiments.mdx [hunk 10]
```diff
@@ -0,0 +1,387 @@
...
+    model: z.string().default('gpt-4o'),
+    maxTokens: z.number().default(500),
+    style: z.enum(['bullet-points', 'paragraph']).default('bullet-points'),
+  }),
+});
+
+const { flag, pickFlags } = createAppScope({ flagSchema });
+
+export { flag, pickFlags };
+```
```
Comment (c-ehrlich) [inline ai-engineering/evaluate/flags-experiments.mdx:60]: Note: we never really discussed the "App scope"/`createAppScope` naming, I just picked it because I couldn't think of anything better at the time, if there are suggestions for something more descriptive I'd be open to that.

File: ai-engineering/evaluate/flags-experiments.mdx [hunk 11]
```diff
@@ -0,0 +1,387 @@
...
+
+### Use flags in your capability
```
Comment (c-ehrlich) [inline ai-engineering/evaluate/flags-experiments.mdx:62]: love this entire section

File: ai-engineering/evaluate/flags-experiments.mdx [hunk 12]
```diff
@@ -0,0 +1,387 @@
...
+    }),
+  });
+
+  return result.object;
+}
+```
+
+### Declare flags in evaluations
+
+Tell your evaluation which flags it depends on using `pickFlags()`:
```
Comment (c-ehrlich) [inline ai-engineering/evaluate/flags-experiments.mdx]: Maybe we need to say what the benefit of this is? 

Basically
- someone else reading the code can know which flags it might use
- if it uses any other flags, the eval will warn you

File: ai-engineering/evaluate/flags-experiments.mdx [hunk 13]
```diff
@@ -0,0 +1,387 @@
+---
+title: "Flags and experiments"
+description: "Use flags to parameterize AI capabilities and run experiments comparing different configurations."
+keywords: ["ai engineering", "flags", "experiments", "configuration", "parameters", "testing"]
+---
+
+import { definitions } from "/snippets/definitions.mdx"
+
+<Tooltip tip={definitions.Flag}>Flags</Tooltip> are configuration parameters that control how your AI capability behaves. By defining flags, you can run <Tooltip tip={definitions.Experiment}>experiments</Tooltip> that systematically compare different models, prompts, temperatures, or architectural approaches - all without changing your code.
```
Comment (c-ehrlich) [inline ai-engineering/evaluate/flags-experiments.mdx]: maybe tool availability (the more tools, the more you pollute your context!), tool behaviour something or other (re temperature replacement)?

File: ai-engineering/evaluate/flags-experiments.mdx [hunk 14]
```diff
@@ -0,0 +1,387 @@
...
+  documentRetrieval: z.object({
+    topK: z.number().default(5),
+    similarityThreshold: z.number().default(0.7),
+  }),
+});
+```
+
+### Set sensible defaults
+
+Choose defaults that work well for most cases. Experiments then test variations:
```
Comment (c-ehrlich) [inline ai-engineering/evaluate/flags-experiments.mdx:239]: Note: for evals that use application code, the default is what the app does in prod!

File: ai-engineering/evaluate/flags-experiments.mdx [hunk 15]
```diff
@@ -0,0 +1,387 @@
...
+}),
+```
+
+### Use enums for discrete choices
+
+When flags have a fixed set of valid values, use enums:
+
+```ts
+// Good: type-safe, clear options
+tone: z.enum(['formal', 'casual', 'friendly']).default('formal'),
```
Comment (c-ehrlich) [inline ai-engineering/evaluate/flags-experiments.mdx:260]: this is also important for something like `model` where passing `string` will type error with the ai sdk

File: ai-engineering/evaluate/flags-experiments.mdx [hunk 16]
```diff
@@ -0,0 +1,387 @@
...
+  --flag.ticketClassification.model=gpt-4o-mini \
+  --flag.ticketClassification.temperature=0.3 \
+  --flag.ticketClassification.systemPrompt=detailed
+
+# Compare against premium configuration
+axiom eval \
+  --flag.ticketClassification.model=gpt-4o \
+  --flag.ticketClassification.temperature=0.3 \
+  --flag.ticketClassification.systemPrompt=detailed
+```
```
Comment (c-ehrlich) [inline ai-engineering/evaluate/flags-experiments.mdx]: this was already mentioned a few paragraphs up

File: ai-engineering/evaluate/flags-experiments.mdx [hunk 17]
```diff
@@ -0,0 +1,387 @@
...
+```bash
+# Test each strategy
+for strategy in chain-of-thought few-shot zero-shot structured-output; do
+  axiom eval --flag.summarization.strategy=$strategy
+done
+```
+
+### Cost vs quality optimization
+
+Find the sweet spot between performance and cost:
```
Comment (c-ehrlich) [inline ai-engineering/evaluate/flags-experiments.mdx:311]: Not necessarily related to _this specific_ section, but how many examples do we need here? Hopefully people can figure out what they can do with parametrizable runtime flags?

File: ai-engineering/evaluate/run-evaluations.mdx [hunk 18]
```diff
@@ -0,0 +1,104 @@
...
+axiom eval --list
+```
+
+## Common options
+
+During development, use watch mode to automatically re-run evaluations when files change:
+
+```bash
+axiom eval --watch
+```
```
Comment (c-ehrlich) [inline ai-engineering/evaluate/run-evaluations.mdx]: i dont know if we should advertise this. watch mode is a lot less useful when it takes 5 minutes to run and costs $3

File: ai-engineering/create.mdx [hunk 19]
```diff
@@ -1,133 +1,159 @@
...
+The features below are experimental. Axiom’s current focus is on the evaluation and observability stages of the AI engineering workflow.
+</Note>
+
+For teams wanting more structure around prompt definitions, Axiom’s SDK includes experimental utilities for managing prompts as versioned objects.
+
+### Define prompts as objects
+
+Represent capabilities as structured `Prompt` objects:
+
+```ts src/prompts/ticket-classifier.prompt.ts
```
Comment (c-ehrlich) [inline ai-engineering/create.mdx:90]: IMO: completely remove prompt management from docs for now. I would like to stop exporting the current version of it from the SDK as well.

File: ai-engineering/iterate.mdx [hunk 20]
```diff
@@ -1,57 +1,147 @@
...
+keywords: ["ai engineering", "iterate", "improvement", "feedback", "annotations"]
 ---
 
-import { Badge } from "/snippets/badge.jsx"
 import { definitions } from '/snippets/definitions.mdx'
+import { Badge } from "/snippets/badge.jsx"
+
+The Iterate stage closes the loop in AI engineering. By analyzing production performance, validating changes through evaluation, and deploying improvements with confidence, you create a continuous cycle of data-driven enhancement.
+
+This is how leading teams build AI capabilities that get better over time.
```
Comment (c-ehrlich) [inline ai-engineering/iterate.mdx]: feels like a weird thing to say in docs imo

File: ai-engineering/concepts.mdx [hunk 21]
```diff
@@ -1,7 +1,7 @@
 ---
 title: "Concepts"
-description: "Learn about the core concepts in AI engineering: Capabilities, Collections, Graders, Evals, and more."
-keywords: ["ai engineering", "AI engineering", "concepts", "capability", "grader", "eval"]
+description: "Learn about the core concepts in AI engineering: Capabilities, Collections, Scorers, Evals, and more."
```
Comment (c-ehrlich) [inline ai-engineering/concepts.mdx]: should feedback go in here also?

File: ai-engineering/concepts.mdx [hunk 22]
```diff
@@ -55,18 +60,26 @@ Ground truth is the validated, expert-approved correct output for a given input.
...
 
-An evaluator, or eval, is the process of testing a capability against a collection of ground truth data using one or more graders. An eval runs the capability on every record in the collection and reports metrics like accuracy, pass-rate, and cost. Evals are typically run before deployment to benchmark performance.
+An evaluator, or eval, is the process of testing a capability against a collection of ground truth data using one or more scorers. An eval runs the capability on every record in the collection and reports metrics like accuracy, pass-rate, and cost. Evals are typically run before deployment to benchmark performance.
 
 ### Online eval
 
-An online eval is the process of applying a grader to a capability’s live production traffic. This provides real-time feedback on performance degradation, cost, and quality drift, enabling continuous monitoring and improvement.
+An online eval is the process of applying a scorer to a capability’s live production traffic. This provides real-time feedback on performance degradation, cost, and quality drift, enabling continuous monitoring and improvement.
 
 ## What’s next?
```
Comment (c-ehrlich) [inline ai-engineering/concepts.mdx:84]: (somehow cant find it in the diff, but noticed this in the deployment:

"Reference
A reference is a historical example of a task completed successfully, serving as a benchmark for AI performance. References provide the input-output pairs that demonstrate the expected behavior and quality standards."

never heard us use that term internally before

File: ai-engineering/quickstart.mdx [hunk 23]
```diff
@@ -98,56 +99,108 @@ To send data to Axiom, configure a tracer. For example, use a dedicated instrume
...
+    };
     ```
 
 For more information on specifying redaction policies, see [Redaction policies](/ai-engineering/redaction-policies).
 
+### Create Axiom configuration file
+
+The Axiom configuration file enables the evaluation framework, allowing you to run systematic tests against your AI capabilities and track improvements over time.
+
+At the root of your project, create the Axiom configuration file `/axiom.config.ts`:
```
Comment (c-ehrlich) [inline ai-engineering/quickstart.mdx:169]: the first paragraph of this page says "Quickly start capturing telemetry data from your generative AI capabilities. After installation and configuration, follow the Axiom AI engineering workflow to create, evaluate, observe, and iterate."

imo we should either remove evals from quickstart, or decide that quickstart includes more than telemetry

File: ai-engineering/create.mdx [hunk 24]
```diff
@@ -1,133 +1,159 @@
...
-The **Create** stage is about defining a new AI <Tooltip tip={definitions.Capability}>capability</Tooltip> as a structured, version-able asset in your codebase. The goal is to move away from scattered, hard-coded string prompts and toward a more disciplined and organized approach to prompt engineering.
+Building an AI <Tooltip tip={definitions.Capability}>capability</Tooltip> starts with prototyping. You can use whichever framework you prefer. Axiom is focused on helping you evaluate and observe your capabilities rather than prescribing how to build them.
+
+TypeScript-based frameworks like Vercel’s [AI SDK](https://sdk.vercel.ai) do integrate most seamlessly with Axiom’s tooling today, but that’s likely to evolve over time.
+
+## Build your capability
+
+Define your capability using your framework of choice. Here’s an example using Vercel's [AI SDK](https://ai-sdk.dev/), which includes [many examples](https://sdk.vercel.ai/examples) covering different capability design patterns. Popular alternatives like [Mastra](https://mastra.ai) also exist.
+
+```ts src/lib/capabilities/classify-ticket.ts expandable
```
Comment (c-ehrlich) [inline ai-engineering/create.mdx:17]: aside: i HATE that we're collapsing code blocks by default. if we want a code block to be collapsed, we should probably just make it shorter instead.

File: ai-engineering/evaluate/write-evaluations.mdx [hunk 25]
```diff
@@ -0,0 +1,316 @@
...
+└── lib/
+    └── capabilities/
+        └── classify-ticket/
+            └── evaluations/
+                ├── spam-classification.eval.ts
+                ├── category-accuracy.eval.ts
+                └── edge-cases.eval.ts
+```
+
+The CLI will find all files matching `**/*.eval.{ts,js,mts,mjs,cts,cjs}` based on your `axiom.config.ts` configuration.
```
Comment (c-ehrlich) [inline ai-engineering/evaluate/write-evaluations.mdx:310]: note: `axiom.config.ts` has `eval.include`/`eval.exclude` to customize these patterns.

**Review by c-ehrlich — COMMENTED** [review]
one general thought: theres a lot of repetition of how to run evals with flags, i think it's covered in at least 3 places.

File: ai-engineering/concepts.mdx (hunk 1 above)
Comment (dominicchapman) [reply ai-engineering/concepts.mdx:41]: We’ll talk more about this one together first before changes. I absolutely want us owning _some_ set of terms around this and you know I was a big fan of these. Prefer to have them landed so the breadth of what we are pursuing is clear. Most examples in docs and repo far too simple at the moment

File: ai-engineering/overview.mdx (hunk 5 above)
Comment (dominicchapman) [reply ai-engineering/overview.mdx:3]: I do actually think so. I think single-turn prompts have their place, but a pain point many teams will/do face is not being able to delve into the more sophisticated types of AI capability because they lack understanding about performance. They’re deemed riskier. Axiom’s features change that.

File: ai-engineering/evaluate/flags-experiments.mdx (hunk 10 above)
Comment (dominicchapman) [reply ai-engineering/evaluate/flags-experiments.mdx:60]: Fair, moving over to Slack. Could argue that introducing `app` language is unnecessary and better substituted with existing language from AI engineering. Let's chat. I have got more used to it though, to be honest.

File: ai-engineering/create.mdx (hunk 24 above)
Comment (dominicchapman) [reply ai-engineering/create.mdx:17]: Hm, not sure I agree on this one. A page can seem very overwhelming with huge code blocks, but the code block itself becomes more relevant once I’m engaging with the material - so expand.  That said, if we can achieve the same in a shorter code block, always for that.

File: ai-engineering/evaluate/analyze-results.mdx [hunk 26]
```diff
@@ -0,0 +1,96 @@
...
+
+```bash
+axiom eval your-eval-name
+# Note the run ID from the output: run_abc123xyz
+```
+
+Make a change (update a prompt, switch models, adjust temperature), then run again with the baseline:
+
+```bash
+axiom eval your-eval-name --baseline run_abc123xyz
```
Comment (islameldigwi) [inline ai-engineering/evaluate/analyze-results.mdx]: what is `run_abc123xyz` ? the CLI expects a baseline id here (trace_id). I think `run_...` might confuse users to think its a run id.

File: ai-engineering/evaluate/run-evaluations.mdx [hunk 27]
```diff
@@ -0,0 +1,104 @@
...
+For quick local testing without sending traces to Axiom, use debug mode:
+
+```bash
+axiom eval --debug
+```
+
+To compare results against a previous run, specify a baseline:
+
+```bash
+axiom eval --baseline <run-id>
```
Comment (islameldigwi) [inline ai-engineering/evaluate/run-evaluations.mdx]: `<run-id>` thats not correct, this expect an eval id, not a run id.

File: ai-engineering/evaluate/setup.mdx [hunk 28]
```diff
@@ -0,0 +1,192 @@
...
+<Note>
+Authenticating with the CLI ensures that evaluation runs are recorded in Axiom and attributed to your user account. This makes it easy to track who ran which experiments and compare results across your team.
+</Note>
+
+### Login with OAuth
+
+Run the login command to authenticate via OAuth:
+
+```bash
+npx axiom auth login
```
Comment (islameldigwi) [inline ai-engineering/evaluate/setup.mdx:46]: we can use `npx axiom login` as its shorter. Both commands work and do the same but I like axiom login better.

File: ai-engineering/concepts.mdx (hunk 4 above)
Comment (dominicchapman) [reply ai-engineering/concepts.mdx]: Good spot - Let's use one term

File: ai-engineering/create.mdx (hunk 19 above)
Comment (dominicchapman) [reply ai-engineering/create.mdx:90]: Going to keep this out of scope for this PR - I don't disagree but not actually sure if we should remove Create altogether - do docs make sense with _nothing_ about it?

File: ai-engineering/quickstart.mdx (hunk 23 above)
Comment (dominicchapman) [reply ai-engineering/quickstart.mdx:169]: @manototh is likely to have thoughts on this - I’ll probably leave this decision to him

File: ai-engineering/evaluate/setup.mdx (hunk 28 above)
Comment (dominicchapman) [reply ai-engineering/evaluate/setup.mdx:46]: Unclear to me if other `auth` commands work with this shorthand. If not, prefer we are consistent

File: ai-engineering/evaluate/write-evaluations.mdx (hunk 25 above)
Comment (dominicchapman) [reply ai-engineering/evaluate/write-evaluations.mdx:310]: Unclear if there is an action here - skipping

**Review by gabrielelpidio — APPROVED** [review]

File: ai-engineering/evaluate/flags-experiments.mdx [hunk 29]
```diff
@@ -0,0 +1,356 @@
...
+import { pickFlags } from '../../../app-scope';
+import { classifyTicket } from '../prompts';
+
+Eval('spam-classification', {
+  // Declare which flags this eval uses
+  configFlags: pickFlags('ticketClassification'),
+  
+  data: () => [],
+  task: async ({ input }) => await classifyTicket(input),
+  scorers: [],
```
Comment (c-ehrlich) [inline ai-engineering/evaluate/flags-experiments.mdx]: i appreciate keeping it simple

but is it better to have empty data/scorers or some simple example here?

Comment (c-ehrlich) [reply ai-engineering/evaluate/flags-experiments.mdx]: in another file we do this

```
Eval('evaluation-name', {
  data: [/* test cases */],
  task: async ({ input }) => {/* run capability */},
  scorers: [/* scoring functions */],
  metadata: {/* optional metadata */},
});
```

File: ai-engineering/evaluate/write-evaluations.mdx [hunk 30]
```diff
@@ -0,0 +1,316 @@
...
+
+## Anatomy of an evaluation
+
+The `Eval` function defines a complete test suite for your capability. Here’s the basic structure:
+
+```ts
+import { Eval, Scorer } from 'axiom/ai/evals';
+
+Eval('evaluation-name', {
+  data: () => [/* test cases */],
```
Comment (c-ehrlich) [inline ai-engineering/evaluate/write-evaluations.mdx]:
```suggestion
  data: [/* test cases */],
```

File: ai-engineering/evaluate/flags-experiments.mdx [hunk 31]
```diff
@@ -0,0 +1,356 @@
...
+```ts src/lib/capabilities/classify-ticket/evaluations/spam-classification.eval.ts
+import { Eval, Scorer } from 'axiom/ai/evals';
+import { pickFlags } from '../../../app-scope';
+import { classifyTicket } from '../prompts';
+
+Eval('spam-classification', {
+  // Declare which flags this eval uses
+  configFlags: pickFlags('ticketClassification'),
+  
+  data: () => [],
```
Comment (c-ehrlich) [inline ai-engineering/evaluate/flags-experiments.mdx]:
```suggestion
  data: [],
```

File: ai-engineering/evaluate/write-evaluations.mdx [hunk 32]
```diff
@@ -0,0 +1,316 @@
...
+
+Create custom scorers using the `Scorer` wrapper:
+
+```ts
+import { Scorer } from 'axiom/ai/evals';
+
+const ExactMatchScorer = Scorer(
+  'exact-match',
+  ({ output, expected }) => {
+    return output.sentiment === expected.sentiment ? 1 : 0;
```
Comment (c-ehrlich) [inline ai-engineering/evaluate/write-evaluations.mdx]:
```suggestion
    return output.sentiment === expected.sentiment ? true : false;
```

File: ai-engineering/evaluate/write-evaluations.mdx [hunk 33]
```diff
@@ -0,0 +1,316 @@
...
+
+Scorers can return just a score, or an object with a score and metadata:
+
+```ts
+const DetailedScorer = Scorer(
+  'detailed-match',
+  ({ output, expected }) => {
+    const match = output.sentiment === expected.sentiment;
+    return {
+      score: match ? 1 : 0,
```
Comment (c-ehrlich) [inline ai-engineering/evaluate/write-evaluations.mdx]:
```suggestion
      score: match ? true : false,
```

File: ai-engineering/evaluate/write-evaluations.mdx [hunk 34]
```diff
@@ -0,0 +1,316 @@
...
+  });
+
+  return result.object;
+}
+
+// Custom scorer for category matching
+const CategoryScorer = Scorer(
+  'category-match',
+  ({ output, expected }) => {
+    return output.category === expected.category ? 1 : 0;
```
Comment (c-ehrlich) [inline ai-engineering/evaluate/write-evaluations.mdx]:
```suggestion
    return output.category === expected.category ? true : false;
```

File: ai-engineering/evaluate/write-evaluations.mdx [hunk 35]
```diff
@@ -0,0 +1,316 @@
...
+  }
+);
+
+// Custom scorer for high-confidence predictions
+const ConfidenceScorer = Scorer(
+  'high-confidence',
+  ({ output }) => {
+    return output.confidence >= 0.8 ? 1 : 0;
```
Comment (c-ehrlich) [inline ai-engineering/evaluate/write-evaluations.mdx]:
```suggestion
    return output.confidence >= 0.8 ? true : false;
```

File: ai-engineering/evaluate/write-evaluations.mdx [hunk 36]
```diff
@@ -0,0 +1,316 @@
...
+  }
+);
+
+// Define the evaluation
+Eval('spam-classification', {
+  data: () => [
```
Comment (c-ehrlich) [inline ai-engineering/evaluate/write-evaluations.mdx]:
```suggestion
  data: [
```

File: ai-engineering/evaluate/write-evaluations.mdx [hunk 37]
```diff
@@ -0,0 +1,316 @@
...
+
+The `data` parameter defines your collection of test cases. Start with a small set of examples and grow it over time as you discover edge cases.
+
+### Inline collections
+
+For small collections, define test cases directly in the evaluation:
+
+```ts
+Eval('classify-sentiment', {
+  data: () => [
```
Comment (c-ehrlich) [inline ai-engineering/evaluate/write-evaluations.mdx]:
```suggestion
  data: [
```

File: ai-engineering/measure.mdx [hunk 38]
```diff
@@ -1,85 +1,234 @@
...
+
+// Custom spam classification scorer
+const SpamClassificationScorer = Scorer(
+  "Spam-Classification",
+  ({ output, expected }: {
+    output: { category: string };
+    expected: { category: string };
+  }) => {
+    const isSpam = (item: { category: string }) => item.category === "spam";
+    return isSpam(output) === isSpam(expected) ? 1 : 0;
```
Comment (c-ehrlich) [inline ai-engineering/measure.mdx:96]:
```suggestion
    return isSpam(output) === isSpam(expected) ? true : false;
```

File: ai-engineering/measure.mdx [hunk 39]
```diff
@@ -1,85 +1,234 @@
...
+  }
+);
+
+// Define the evaluation
+Eval('spam-classification', {
+  // Specify which flags this eval uses
+  configFlags: pickFlags('ticketClassification'),
+  
+  // Test data with input/expected pairs
+  data: () => [
```
Comment (c-ehrlich) [inline ai-engineering/measure.mdx]:
```suggestion
  data: [
```
